viite-UI/src/model/RoadAddressProjectCollection.js

```diff
@@ -142,7 +142,15 @@
         name: data[0].value,
         startDate: data[1].value,
         additionalInfo: data[2].value,
-        roadPartList: dirtyRoadPartList
+        roadPartList: _.map(dirtyRoadPartList, function(part){
```

Comment:
Maybe to have only one list we need to have 2 booleans, because we need to see if it's a dirty deleted or a dirty new added.